make/tools.cmake

```diff
@@ -0,0 +1,43 @@
+function(DIFF)
+  # ~~~
+  # DIFF(
+  # NAME
+  # GOLDEN
+  # ACTUAL
+  # )
+  set(oneValueArgs NAME GOLDEN ACTUAL)
+  cmake_parse_arguments(
+    DIFF
+    ""
+    "${oneValueArgs}"
+    ""
+    ${ARGN}
+    )
+
+  set(DIFF_FILE_A ${DIFF_GOLDEN})
+  set(DIFF_FILE_B ${DIFF_ACTUAL})
+
+  get_file_target(DIFF_FILE_A_TARGET ${DIFF_FILE_A})
```

Comment:
Use append_file_dependency over directly depending on get_file_target.  It solves transitive dependencies.